core/client/src/main/java/alluxio/client/file/UnknownLengthFileInStream.java

```diff
@@ -122,6 +124,8 @@ protected BlockInStream createUnderStoreBlockInStream(long blockStart, long leng
     return new UnderStoreBlockInStream(blockStart, Constants.UNKNOWN_SIZE, length, path);
   }
 
+  // TODO(peis): I don't see why this is necessary. Ask Gene to confirm.
```

Comment:
this was required since the cache stream should only be closed (and not cancelled) if the block was fully read (and also fully written out to the cache stream)